### PR TITLE
[Java] Fixes Spring generator devaultValues for int64/float/double

### PR DESCRIPTION
**Pull request by spacether**
This PR fixes https://github.com/OpenAPITools/openapi-generator/issues/1226
That issue describes that when we have an in64, float, or double CodegenParameter, the string defaultValue includes unneeded suffix characters l,f, or d. Those suffixes were added by AbstractJavaCodegen because that generator assumed that those default values would be used as java values of the correct type. Instead, the spring generator uses String type default values for CodegenParameter only. Typically those values are handled with parseLong/parseFloat/parseDouble etc which fail if the value includes a character suffix.

This update keeps the suffixes on CodegenProperties because templates use those defaultValues as Long/Float/Double literal values.
CodegenProperty defaultValues:
- "9223372036854775807l" (int64)
- "3.14159f" (float)
- "3.14159d" (double)

For CodegenParameter we now use values which lack the suffix so parseLong/parseDouble/parseFloat will work
CodegenParameter defaultValues:
- "9223372036854775807" (int64)
- "3.14159" (float)
- "3.14159" (double)

### Further work
I verified that all spring generators use classes that require String defaultValues for query params.
These are:
1. spring-mvc-petstore-server (ApiParam,RequestParam String defaultValue)
2. spring-cloud-feign-petstore (ApiParam, RequestParam String defaultValue)
3. spring-cloud-feign-async-petstore (ApiParam, RequestParam String defaultValue)
4. spring-stubs (ApiParam, RequestParam String defaultValue)
5. spring-mvc-petstore-j8-async-server (ApiParam, RequestParam String defaultValue)
6. spring-mvc-petstore-j8-localdatetime (ApiParam, RequestParam String defaultValue)
7. spring-delegate (ApiParam, RequestParam String defaultValue)
8. spring-delegate-j8 (ApiParam, RequestParam String defaultValue)
9. springboot-petstore-server (ApiParam, RequestParam String defaultValue)
10. springboot-petstore-server-reactive (ApiParam, RequestParam String defaultValue)
11. springboot-petstore-server-beanvalidation (ApiParam, RequestParam String defaultValue)
12. springboot-petstore-server-implicitHeaders (ApiParam, RequestParam String defaultValue)
13. springboot-petstore-server-useOptional (ApiParam, RequestParam String defaultValue)
14. springboot-virtualan-petstore-server (ApiParam, RequestParam String defaultValue)

### Verification Test
- A test has been added at https://github.com/spacether/openapi-generator/blob/issue_1226_fix_defaultValue/modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java#L485

### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [ ] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

Java Technical Committee:
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java [hunk 1]
```diff
@@ -849,4 +852,83 @@ public void setPerformBeanValidation(boolean performBeanValidation) {
...
+                    typeDeclaration = "";
+                }
+            }
+
+            return String.format(Locale.ROOT, pattern, typeDeclaration);
+        } else if (ModelUtils.isIntegerSchema(p)) {
+            if (p.getDefault() != null) {
+                return p.getDefault().toString();
+            }
+            return null;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java]: We need a case for Long (isLongSchema) and add "L" at the end of the default value. Please see the error message below for details.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java [hunk 2]
```diff
@@ -849,4 +852,83 @@ public void setPerformBeanValidation(boolean performBeanValidation) {
...
+        } else if (ModelUtils.isNumberSchema(p)) {
+            if (p.getDefault() != null) {
+                return p.getDefault().toString();
+            }
+            return null;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java]: We need a case for Float (isFloatSchema) and add "F" at the end of the default value. Please see the error message below for details.

File: modules/openapi-generator/src/test/resources/2_0/issue1226.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      description: ''
+      operationId: numericQueryParams
+      produces:
+        - application/json
+      parameters:
+        - in: query
+          name: int64
+          type: integer
+          format: int64
+          default: 9223372036854775807
```
Comment (wing328) [inline modules/openapi-generator/src/test/resources/2_0/issue1226.yaml]: FYI. I got the following when testing the spring app generated from this spec:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project openapi-spring: Compilation failure
[ERROR] /private/tmp/spring/src/main/java/org/openapitools/model/NumberHolder.java:[20,24] integer number too large: 9223372036854775807
```

Appended "L" at the end of the default value fixes it.

For float, I fixed the issue by appending "F" to the default value.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java (hunk 1 above)
Comment (spacether) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java]: For CodegenProperty, I changed the code back to preserve the "l" suffix for this case. If you need me to change our existing code in AbstractJavaCodegen.java from adding "l" to "L" please let me know.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java (hunk 2 above)
Comment (spacether) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java]: For CodegenProperty, I changed the code back to preserve the "f" suffix for this case. If you need me to change our existing code in AbstractJavaCodegen.java from adding "f" to "F" please let me know.

File: modules/openapi-generator/src/test/resources/2_0/issue1226.yaml (hunk 3 above)
Comment (spacether) [reply modules/openapi-generator/src/test/resources/2_0/issue1226.yaml]: For CodegenProperty, I changed the code back to preserve the "l" suffix for this case. If you need me to change our existing code in AbstractJavaCodegen.java from adding "l" to "L" please let me know.

For CodegenProperty, I changed the code back to preserve the "f" suffix for this case. If you need me to change our existing code in AbstractJavaCodegen.java from adding "f" to "F" please let me know.

**Comment by wing328** [comment]
Tested again and no longer encounter the issue.

**Comment by fooo** [comment]
Exactly the same problem occurs with the **jaxrs-cxf** generator. In the templates_@DefalutValue_ annotations are generated based on the model _defaultvalue_ parameter. This results in a @DefaultValue with a postfix which results in invalid default values.

**Comment by wing328** [comment]
@fooo I wonder if you can file a PR with a similar fix.

**Pull request merged**